### PR TITLE
Docs/matching demo

### PR DESCRIPTION
**Pull request by soanvig**
https://origin-247-matching-demo.surge.sh

### CODE REVIEW

File: packages/matching-demo/src/algorithm/spreadMatcher.ts [hunk 1]
```diff
@@ -0,0 +1,302 @@
+import { omit, cloneDeep } from 'lodash';
```
Comment (soanvig) [inline packages/matching-demo/src/algorithm/spreadMatcher.ts:1]: This is copy-paste from algorith running in backend, with one more value returned from the function for debugging purposes

File: packages/matching-demo/src/algorithm/spreadMatcher.ts [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+                match1.entities[0] === match2.entities[0] &&
+                match1.entities[1] === match2.entities[1]
+            );
+        };
+
+        return matches.reduce((sum, match) => {
+            const existingMatch = sum.find((s) => areMatchesSame(s, match));
+
+            if (existingMatch) {
+                existingMatch.volume = existingMatch.volume.add(match.volume);
```
Comment (alexworker23) [inline packages/matching-demo/src/algorithm/spreadMatcher.ts:156]: What is the purpose of this assignment? 
As far as I understand it is not mutating the `existingMatch`? 
and you simply return the `sum`

**Review by alexworker23 — APPROVED** [review]

File: packages/matching-demo/src/algorithm/spreadMatcher.ts (hunk 2 above)
Comment (soanvig) [reply packages/matching-demo/src/algorithm/spreadMatcher.ts:156]: Don't dive into this algorithm please ;) This function is not clean TBH, I must agree (it does reduce + sideeffect), I may look into it when I will be touching algo soon

File: packages/matching-demo/src/App.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            </Typography>
+            {roundMatches.length === 0 && 'No matches'}
+            {roundMatches.map((_, i) => (
+                <MatchRound
+                    consumption={consumers}
+                    generation={generators}
+                    allMatches={roundMatches.slice(0, i + 1).flat()}
+                    roundMatches={roundMatches[i]}
+                    round={i + 1}
+                    key={i}
```
[the remark this answers is not in the document]
Comment (soanvig) [reply packages/matching-demo/src/App.tsx:54]: Yeah, I kindof knows that, but in this case there is no other thing to describe matching round (key), than it's position in array.

File: packages/matching-demo/src/App.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import Typography from '@material-ui/core/Typography';
+import Container from '@material-ui/core/Container';
+import { PriorityConfigurator } from './components/PriorityConfigurator';
+import { IPriority } from './types';
+import './App.css';
+
+export function App() {
+    const [consumers, setConsumers] = useState([] as { id: string; volume: BigNumber }[]);
+    const [generators, setGenerators] = useState([] as { id: string; volume: BigNumber }[]);
+    const [priority, setPriority] = useState([] as IPriority[][]);
```
[the remark this answers is not in the document]
Comment (soanvig) [reply packages/matching-demo/src/App.tsx:15]: Personal preference. My opinion on that is that I don't want to mess visually pair `behavior (useState) - value ([])` by inserting type in the middle

**Comment by soanvig** [comment]
force push with squashed commits

**Pull request merged**